kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/OpenGaussWALDumper.java

```diff
@@ -138,28 +152,61 @@ private void dump() throws SQLException {
         }
     }
     
+    private int getMajorVersion() throws SQLException {
+        StandardPipelineDataSourceConfiguration dataSourceConfig = (StandardPipelineDataSourceConfiguration) dumperContext.getCommonContext().getDataSourceConfig();
+        try (
+                Connection connection = DriverManager.getConnection(dataSourceConfig.getUrl(), dataSourceConfig.getUsername(), dataSourceConfig.getPassword());
+                Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery("SELECT version()")) {
+            resultSet.next();
+            String versionText = resultSet.getString(1);
+            return parseMajorVersion(versionText);
+        }
+    }
+    
+    private int parseMajorVersion(final String versionText) {
+        Matcher matcher = VERSION_PATTERN.matcher(versionText);
+        if (matcher.find()) {
+            log.info("openGauss major version={}, `select version()`={}", matcher.group(1), versionText);
+            return Integer.parseInt(matcher.group(1));
+        }
+        return DEFAULT_VERSION;
+    }
+    
     private PgConnection getReplicationConnectionUnwrap() throws SQLException {
         return logicalReplication.createConnection((StandardPipelineDataSourceConfiguration) dumperContext.getCommonContext().getDataSourceConfig()).unwrap(PgConnection.class);
     }
     
-    private void processEventWithTX(final AbstractWALEvent event) {
+    private void processEventWithTX(final AbstractWALEvent event, final int majorVersion) {
         if (event instanceof BeginTXEvent) {
+            if (majorVersion < 3) {
+                return;
+            }
+            if (!rowEvents.isEmpty()) {
+                channel.push(rowEvents.stream().map(walEventConverter::convert).collect(Collectors.toList()));
+                rowEvents = new LinkedList<>();
+            }
```

Comment:
`rowEvents` belongs to previous transaction, and BeginTXEvent belongs to next transaction, looks it should not occur if CommitTXEvent parsing works.

Could we remove this block of code and just `log.warn`?